### ISSUE TITLE
Error Code 139 thrown before test run When Using Unity 2020.1.0f1

### ISSUE THREAD

**Issue by Alriightyman** [webbertakken/unity-test-runner#67]
I've been trying to get tests to run but came across this error:
```
###########################
#   Testing in EditMode   #
###########################

Segmentation fault (core dumped)
[Licensing::Module] Channel doesn't exist: "LicenseClient-root"
[Licensing::Module] Successfully launched the LicensingClient (PId: 81)
[SignatureVerifier] Application signature verification not supported on this platform.
[Licensing::Module] Successfully connected to LicensingClient on channel: "LicenseClient-root" (connect: 0.81s, validation: 0.07s, handshake: 0.00s)
[Licensing::Module] Connected to LicensingClient (PId: 81, launch time: 0.00, total connection time: 0.88s)
Entitlement-based licensing initiated
[LicensingClient] Licenses updated successfully
Unexpected exit code 139

###########################
#   Testing in PlayMode   #
###########################

Segmentation fault (core dumped)
[Licensing::Module] Channel doesn't exist: "LicenseClient-root"
[Licensing::Module] Successfully launched the LicensingClient (PId: 136)
[SignatureVerifier] Application signature verification not supported on this platform.
[Licensing::Module] Successfully connected to LicensingClient on channel: "LicenseClient-root" (connect: 0.57s, validation: 0.07s, handshake: 0.00s)
[Licensing::Module] Connected to LicensingClient (PId: 136, launch time: 0.00, total connection time: 0.64s)
Entitlement-based licensing initiated
[LicensingClient] Licenses updated successfully
Unexpected exit code 139
```
Seems to be a Seg Fault error. A way to fix this was mentioned here:
https://gitlab.com/gableroux/unity3d-gitlab-ci-example/-/issues/121
This suggests adding the `-nographics` flag to all CI scripts. 

I would assume adding that flag to this file:
`unity-test-runner/action/steps/run_tests.sh`
when executing the tests would fix the issue.

I am unsure as to why you would want to run this *with* graphics, however, I am new to running CI and Unity in this way so maybe there is a reason for this flag to be omitted from those calls?  If so, maybe a variable that can be set and passed?  What are your thoughts?

**Comment by webbertakken** [comment]
@Alriightyman thank you for reporting your issue and especially for having done some research.

That might be a fix but we need to be sure that it doesn't break anything else.

> I am unsure as to why you would want to run this with graphics, however, I am new to running CI and Unity in this way so maybe there is a reason for this flag to be omitted from those calls? If so, maybe a variable that can be set and passed? What are your thoughts?

The reason nographics is omitted is that we actually emulate graphics. In unity builder the nographics flag is also not given: https://github.com/webbertakken/unity-builder/blob/master/action/steps/build.sh#L111

I'm not sure if adding the `-nographics` flag is a good solution, because tests use playmode to run and as far as I know playmode needs graphics (though i'm not 100% certain about this).

I suspect that this is actually an issue with `2020.1.0f1` as I saw more people with similar issues, also in gitlab and the same issue does not occur for other versions of unity. I guess we also need to find out whether we can do with that flag or not.

**Comment by Alriightyman** [comment]
I see. I forked your repo and made those changes. It does indeed pass now but I'm using default tests. If you can give me some type of test that you would expect to not work correctly with those changes, I can try them out and see what happens.

**Comment by webbertakken** [comment]
Cool. Personally I don't have any ready to go examples.

As stated above, my opinion is that it's likely an issue with the image or a change in this unity version, that we should identify instead of just adding a flag that was never needed in other versions of unity. 

As the official documentation states, it doesn't allow global illumination so I think at least for builder this won't be a viable option.

**Comment by ncipollo** [comment]
Hi! I've run into the same issue with the slightly more recent `2020.1.2f1` image. I tried out Alriightyman's fork and it also resolved the issue for me in my project.

The CLI documentation for unity does call this flag out specifically for CI -
> Note: If you use continuous integration (CI) tools like Jenkins to activate via the command line, add the -nographics flag to prevent a WindowServer error.
(From https://docs.unity3d.com/2020.2/Documentation/Manual/CommandLineArguments.html)

I suspect that the 2020 version of Unity does not gracefully handle batch mode CLI requests on machines which are running headless without this flag. I wonder if it would make sense to at least introduce an action argument to utilize this flag until the issue gets cleared up in either the docker image or unity? If so, I'd be happy to help.

**Comment by webbertakken** [comment]
@ncipollo please not that the same documentation also states

> -nographics does not allow you to bake GI as Enlighten requires GPU acceleration.

Please read https://github.com/webbertakken/unity-test-runner/issues/67#issuecomment-668825769 above. 

I'm not in favour of "just adding" a flag that might cause backward compatibility issues in the future. There should be a good rationale for adding it, as all images before 2020 worked perfectly without it.

Help is more than welcome.

**Comment by ncipollo** [comment]
@webbertakken,
Yep, saw that comment and it totally makes sense, though I do wonder if the containers which the github actions run within would be able to utilize GPU acceleration even if `-nographics` is not supplied.

**Comment by webbertakken** [comment]
> @webbertakken,
> Yep, saw that comment and it totally makes sense, though I do wonder if the containers which the github actions run within would be able to utilize GPU acceleration even if `-nographics` is not supplied.

Entirely valid question!

Honestly I don't know too much about unity. I created the actions before I started my first project, lol. I started working on my first unity project this year. So it's really up to the community to determine what's right for test-runner and builder. I'll just act as a sanity checker :)

**Comment by webbertakken** [comment]
Possibly related known bugs since `2020.1.0`

- Global Illumination: Crash with empty stacktrace when starting bake in the new scene after baking previous scene with GPU PLM ([1244384](https://issuetracker.unity3d.com/issues/crash-with-empty-stacktrace-when-starting-bake-in-the-new-scene-after-baking-previous-scene-with-gpu-plm))
- Global Illumination: [macOS] BugReporter doesn't get invoked when the project crashes ([1219458](https://issuetracker.unity3d.com/issues/macos-bugreporter-doesnt-get-invoked-when-the-project-crashes))
- Global Illumination: gi::InitializeManagers() takes 0.6s during Editor startup ([1162775](https://issuetracker.unity3d.com/issues/gi-initializemanagers-takes-0-dot-4s-during-editor-startup))

**Comment by webbertakken** [comment]
Dug through all the official documentation and 2020 change logs related to this topic. Apart from the above post there doesn't seem  to be anything published that is related to this problem.

Looking at the raw logs from a Unity Cloud Build, it doesn't appear to be using the `-nographics` flag.
I could only check this for 2019 builds, as I currently no longer have the subscription.

Perhaps someone else could confirm this for a build using Unity 2020.1.x.
